testing/tests/consensus_task.rs

```diff
@@ -1,346 +1,344 @@
-//                 } else {
-//                     state.handle_event(event).await;
-//                     (None, state)
-//                 }
-//             }
-//             .boxed()
-//         },
-//     ));
-//     let consensus_name = "Consensus Task";
-//     let consensus_event_filter = FilterEvent::default();
-//
-//     let consensus_task_builder = TaskBuilder::<
-//         ConsensusTaskTypes<TYPES, I, HotShotSequencingConsensusApi<TYPES, I>>,
-//     >::new(consensus_name.to_string())
-//     .register_event_stream(event_stream.clone(), consensus_event_filter)
-//     .await
-//     .register_registry(&mut registry.clone())
-//     .await
-//     .register_state(consensus_state)
-//     .register_event_handler(consensus_event_handler);
-//     // impossible for unwrap to fail
-//     // we *just* registered
-//     let consensus_task_id = consensus_task_builder.get_task_id().unwrap();
-//     let consensus_task = ConsensusTaskTypes::build(consensus_task_builder).launch();
-//
-//     task_runner.add_task(
-//         consensus_task_id,
-//         consensus_name.to_string(),
-//         consensus_task,
-//     )
-// }
+use commit::Committable;
+use either::Right;
+use hotshot::traits::NodeImplementation;
+use hotshot::types::SystemContextHandle;
+use hotshot::{certificate::QuorumCertificate, traits::TestableNodeImplementation, SystemContext};
+
+use hotshot::rand::SeedableRng;
+
+use hotshot::tasks::add_consensus_task;
+
+use hotshot::traits::Block;
+use hotshot::types::SignatureKey;
+use hotshot::HotShotInitializer;
+use hotshot::HotShotSequencingConsensusApi;
+use hotshot_consensus::traits::ConsensusSharedApi;
+use hotshot_task::event_stream::ChannelStream;
+
+use hotshot_task_impls::events::SequencingHotShotEvent;
+use hotshot_testing::node_types::SequencingMemoryImpl;
+use hotshot_testing::node_types::SequencingTestTypes;
+use hotshot_testing::test_builder::TestMetadata;
+use hotshot_types::message::GeneralConsensusMessage;
+use hotshot_types::message::Proposal;
+use jf_primitives::signatures::BLSSignatureScheme;
+
+use hotshot_types::data::QuorumProposal;
+use hotshot_types::data::SequencingLeaf;
+use hotshot_types::data::ViewNumber;
+use hotshot_types::message::Message;
+
+use hotshot_types::traits::election::Membership;
+
+use hotshot_types::traits::election::QuorumExchangeType;
+use hotshot_types::traits::election::SignedCertificate;
+use hotshot_types::traits::metrics::NoMetrics;
+
+use hotshot_types::traits::node_implementation::CommitteeEx;
+use hotshot_types::traits::node_implementation::ExchangesType;
+
+use hotshot_types::traits::node_implementation::QuorumEx;
+
+use hotshot::traits::election::vrf::JfPubKey;
+use hotshot_task_impls::harness::run_harness;
+use hotshot_types::traits::signature_key::EncodedSignature;
+use hotshot_types::traits::{
+    election::ConsensusExchange, node_implementation::NodeType, state::ConsensusTime,
+};
+
+use std::collections::HashMap;
+
+async fn build_consensus_api(
```

Comment:
I'll probably move this function to a separate file and refactor it a bit, so it can be used for other tasks as well. Just a heads-up so you won't think it's disappeared. 😃